cpp/oneapi/dal/table/detail/csr_accessor.hpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/detail/sparse_access_iface.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+/// @tparam T The type of data values in blocks returned by the accessor.
+///           Should be const-qualified for read-only access.
+///           An accessor supports at least :expr:`float`, :expr:`double`, and :expr:`std::int32_t` types of :literal:`T`.
+template <typename T>
+class csr_accessor {
+public:
+    using data_t = std::remove_const_t<T>;
+    static constexpr bool is_readonly = std::is_const_v<T>;
+
+    /// Creates a read-only accessor object from the csr table. Available only for
+    /// const-qualified :literal:`T`.
+    template <typename U = T, std::enable_if_t<std::is_const_v<U>, int> = 0>
+    explicit csr_accessor(const table& table)
```

Comment:
Are you sure that every table shall support this accessor?

Reply:
It was fixed, only for csr_table